core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -556,6 +556,8 @@ private LoadTableResponse stageCreate() {
   private static List<MetadataUpdate> createChanges(TableMetadata meta) {
     ImmutableList.Builder<MetadataUpdate> changes = ImmutableList.builder();
 
+    changes.add(new MetadataUpdate.UpgradeFormatVersion(meta.formatVersion()));
```

Comment:
I think this should be set only if the format version is not the default?

Reply:
I think the server-side default and the client-side default may not match? That is the issue I originally encountered, the Trino plugin had v2 as the default and set the initial sequence number to 1, but the server didn't receive a format version so treated it as v1, and failed because the sequence number was > 0.

Reply:
Okay, I see what you mean.

Reply:
+1. But I too see what you mean.

I could have sworn I left an almost identical comment when I reviewed this before. I can possibly help with adding a unit test for this sequence in `CatalogTests` if we’d like. Feel free to reach out on Slack too.